src/transforms/bin.js

```diff
@@ -184,9 +184,10 @@ function binn(
               groupFacet.push(i++);
               groupData.push(reduceData.reduceIndex(b, data, extent));
               if (K) GK.push(k);
-              if (Z) GZ.push(G === Z ? f : Z[b[0]]);
-              if (F) GF.push(G === F ? f : F[b[0]]);
-              if (S) GS.push(G === S ? f : S[b[0]]);
+              const u = b.length === 0 ? g[0] : b[0];
```

Comment:
If we always use `g[0]` here instead of `b[0]`, the tests still pass. So, should we do that? Or can we add a test for this case. I’m struggling to think through the implications here.

Reply:
I added a test to understand the implication here. 👍